e2e/solve/fund.go

```diff
@@ -16,39 +19,92 @@ import (
 	"cosmossdk.io/math"
 )
 
-func maybeFundSolver(ctx context.Context, network netconf.ID, backends ethbackend.Backends) error {
+// erc20 tokens to fund solver with on devnet. useful for solvernet development when forking public networks.
+var testTokenAddressesToFund = []struct {
+	chainID uint64
+	addr    common.Address
+}{
+	// holesky wstETH
+	{chainID: evmchain.IDMockL1, addr: common.HexToAddress("0x8d09a4502cc8cf1547ad300e066060d043f6982d")},
+	// holesky stETH
+	{chainID: evmchain.IDMockL1, addr: common.HexToAddress("0x3f1c547b21f65e10480de3ad8e19faac46c95034")},
+}
+
+func maybeFundERC20Solver(ctx context.Context, network netconf.ID, backends ethbackend.Backends) error {
 	// funding solver with l1 wsETH uses anvil_setStorageAt util, which is only available on devnet
 	if network != netconf.Devnet {
 		return nil
 	}
 
-	// erc20 tokens to fund solver with on devnet. useful for solvernet development when forking public networks
-	toFund := []struct {
-		chainID uint64
-		addr    common.Address
-	}{
-		// holesky wstETH
-		{chainID: evmchain.IDMockL1, addr: common.HexToAddress("0x8d09a4502cc8cf1547ad300e066060d043f6982d")},
-		// holesky stETH
-		{chainID: evmchain.IDMockL1, addr: common.HexToAddress("0x3f1c547b21f65e10480de3ad8e19faac46c95034")},
-	}
-
 	solver := eoa.MustAddress(netconf.Devnet, eoa.RoleSolver)
-	eth1m := math.NewInt(1_000_000).MulRaw(params.Ether).BigInt()
 
-	for _, tkn := range toFund {
+	for _, tkn := range testTokenAddressesToFund {
 		ethCl, ok := backends.Clients()[tkn.chainID]
 		if !ok {
 			continue
 		}
 
+		eth1m := math.NewInt(1_000_000).MulRaw(params.Ether).BigInt()
+
 		// if devnet is not forking the public chain tkn is deployed on, this sets
 		// storage on an unused address, which is fine
-		err := anvil.FundERC20(ctx, ethCl, tkn.addr, eth1m, solver)
-		if err != nil {
+		if err := anvil.FundERC20(ctx, ethCl, tkn.addr, eth1m, solver); err != nil {
 			return errors.Wrap(err, "fund tkn failed", "chain_id", tkn.chainID, "addr", tkn.addr)
 		}
+
+		if err := checkAccountBalance(ctx, ethCl, solver, tkn.addr, tkn.chainID); err != nil {
```

Comment:
not sure why you check native balance after doing token funding (not native funding)?